lib/internal/Magento/Framework/Reflection/TypeProcessor.php

```diff
@@ -521,14 +521,87 @@ public function getParamType(ParameterReflection $param)
                 $param->getDeclaringFunction()->name
             ));
         }
-        if ($type == 'array') {
+        if ($type === 'array') {
             // try to determine class, if it's array of objects
             $paramDocBlock = $this->getParamDocBlockTag($param);
             $paramTypes = $paramDocBlock->getTypes();
             $paramType = array_shift($paramTypes);
+
+            $paramType = $this->resolveFullyQualifiedClassName($param->getDeclaringClass(), $paramType);
+
             return strpos($paramType, '[]') !== false ? $paramType : "{$paramType}[]";
         }
-        return $type;
+
+        return $this->resolveFullyQualifiedClassName($param->getDeclaringClass(), $type);
+    }
+
+    /**
+     * Resolve fully qualified type name in the class alias context
+     *
+     * @param ClassReflection $sourceClass
+     * @param string $typeName
+     * @return string
+     */
+    public function resolveFullyQualifiedClassName(ClassReflection $sourceClass, string $typeName): string
```

Comment:
Is there any specific reason we would like to have this method `public` in this exact class? I would either suggest making it `private` or extracting to a separate class if there are other clients.

```suggestion
    private function resolveFullyQualifiedClassName(ClassReflection $sourceClass, string $typeName): string
```

Reply:
Definitely no reason. It should be private.